src/index.js

```diff
@@ -0,0 +1,87 @@
+const express = require('express')
+const bodyParser = require('body-parser')
+
+const config = require('./config')
+const UserService = require('./services/user')
+const Db = require('./db')
+
+const db = new Db(config.db)
+const userService = new UserService(db)
+const userRouter = express.Router()
+const app = express()
+app.use(bodyParser.json())
+
+/**
+ * @api {get} /user/:id Read the data of a user
+ * @apiName GetUser
+ * @apiGroup User
+ *
+ * @apiParam {Number} id Users unique ID.
+ *
+ * @apiSuccess {String} name          name of the user
+ * @apiSuccess {String} dob           the users date of birth in the format YYYY-MM-DD 
+ * @apiSuccess {String} address       the users address
+ * @apiSuccess {String} description   a description of the user
+ */
+userRouter.get('/:id', (req, res) => {
+  userService.get(req.params.id)
+    .then((user) => {
+      if (user) {
+        res.json(user)
+      } else {
+        res.status(200).send('User not found')
+      }
+    }).catch(e => res.status(400).send(e.message))
+})
+
+/**
+ * @api {post} /user Create a new user
+ * @apiName CreateUser
+ * @apiGroup User
+ *
+ * @apiParam {String} name          the name of the user
+ * @apiParam {String} address       the users address
+ * @apiParam {String} dob           the users date of birth in the format YYYY-MM-DD
+ * @apiParam {String} description   the users description
+ *
+ * @apiSuccess {String} id          the id of the created user
+ */
+userRouter.post('/', (req, res) => {
+  userService.create(req.body)
+    .then(({ id }) => res.json({ id }))
+    .catch(e => res.status(400).send(e.message))
+})
+
+/**
+ * @api {delete} /user/:id Delete a user
+ * @apiName DeleteUser
+ * @apiGroup User
+ *
+ * @apiParam {String} id the id of the user to delete
+ */
+userRouter.delete('/:id', (req, res) => {
+  userService.delete(req.params.id)
+    .then((a) => res.status(200).end())
+    .catch((e) => res.status(400).send(e.message))
+})
+
+/**
+ * @api {post} /user/:id Update an existing user
+ * @apiName UpdateUser
+ * @apiGroup User
+ *
+ * @apiParam {String} [name]          the name of the user
+ * @apiParam {String} [address]       the users address
+ * @apiParam {String} [dob]           the users date of birth in the format YYYY-MM-DD
+ * @apiParam {String} [description]   the users description
+ */
+userRouter.post('/:id', (req, res) => {
```

Comment:
Do you see any other options of http verb for the user updating?
Could you let me know why you chose to use `post ` here?

Reply:
Yes I could have used `PUT` or `PATCH`, some say `PATCH` should only update a single field so possibly `PUT` is more applicable (because we can update multiple fields at a time). 

Again it was just a time constraint thing, I didn't think too much about it.